src/gc/boehm.cr

```diff
@@ -74,6 +74,39 @@ fun __crystal_realloc(ptr : Void*, size : UInt32) : Void*
   LibGC.realloc(ptr, size)
 end
 
+# :nodoc:
+fun __crystal_malloc64(size : UInt64) : Void*
+  {% if flag?(:bits32) %}
+    if size > UInt32::MAX
+      raise ArgumentError.new("Given size is bigger than UInt32::MAX")
```

Comment:
Maybe it make sense to just return `nil` here. malloc may return nil to signal that there is no enough free continuous memory to allocate. Although there is a hard limit here, since the codegen is unable to use this yet, we could at least handle the `nil` in the generated code at `generic_malloc(type)`. WDYT?

Reply:
You mean, `Pointer(Void).new(0)`? That will be a segfault after use.

I think these comments, and some of the last fixes, are a bit unrelated to the original issue. I don't want to keep adding unrelated changes.

Reply:
I disagree @bcardiff, I think malloc/realloc should always return a pointer to a valid memory zone. To ensure that, it should raise an ArgumentError on bad argument (like here), or when not enough memory (some discussions about that in #4345)

Reply:
Yes @asterite a `NULL` or `Pointer(Void).new(0)`. I meant to return `NULL` as a better behavior that could be handled without changing the llvm calls for llvm invokes as suggested https://github.com/crystal-lang/crystal/pull/4960#discussion_r138633469 . As is, the exceptions won't be handled as per the linked comment.

I'm ok to defer this until #4345, I was trying to avoid adding code that seems to won't be actually used as expected.

Reply:
For sure handling the `NULL` requires changes in `generic_malloc(type)`. Let's wait for #4345 then to change the call to invokes probably then and leave more code in crystal and less in the llvm builder.